backend/src/services/session.service.js

```diff
@@ -0,0 +1,69 @@
+const sessionsByID = new Map();
+const sessionsByCode = new Map();
+const { v4: uuidv4 } = require('uuid');
+
+const createSession = (users = []) => {
+    const sessionID = uuidv4();
+    const createdAt = new Date().toISOString();
+
+    let sessionCode = uuidv4().substring(0, 4).toUpperCase();
+    matchingCode = sessionsByCode.get(sessionCode);
+
+    while (matchingCode) {
+        matchingCode = sessionsByCode.get(sessionCode);
+        sessionCode = uuidv4().substring(0, 4).toUpperCase();
```

Comment:
The generation of `sessionCode` must come before `matchingCode`, the current order of these operations doesn't 100% guarantee a unique session code.

Alternative: Instead of changing the order, you can just remove the whole `sessionCode` too, and just put `sessionsByCode.get(sessionCode)` as the codition for while loop as we aren't using that `matchingCode` anywhere again.

Reply:
Pushed changes